File: sdk/examples/hello_aethermind.js
/**
 * Hello AetherMind! 🚀
 * Simple example demonstrating AetherMind AGI integration with JavaScript
 */

const { AetherMindClient } = require('@aethermind/sdk');

async function main() {
  console.log('='.repeat(60));
  console.log('🧠 Hello AetherMind - JavaScript SDK Demo');
  console.log('='.repeat(60));
  console.log();

  // Get API key from environment or use demo
  const apiKey = process.env.AETHERMIND_API_KEY || 'am_live_demo_key';

  if (apiKey === 'am_live_demo_key') {
    console.log('⚠️  Using demo API key. Set AETHERMIND_API_KEY environment variable');
    console.log('   to use your own key from https://aethermind.ai');
    console.log();
  }

  // Initialize client
  console.log('📡 Connecting to AetherMind AGI...');
  let client;
  try {
    client = new AetherMindClient({ apiKey });
    console.log('✅ Connected!\n');
  } catch (error) {
    console.error(`❌ Connection failed: ${error.message}`);
    return;
  }

  // Example 1: Basic chat
  console.log('='.repeat(60));
  console.log('Example 1: Basic Chat');
  console.log('='.repeat(60));

  const question = 'What is the difference between AI and AGI?';
  console.log(`💬 Question: ${question}`);
  console.log();

  try {
    const response = await client.chat({ message: question });
    console.log(`🤖 AetherMind: ${response.answer}`);

    if (response.confidence) {
      console.log(`📊 Confidence: ${(response.confidence * 100).toFixed(0)}%`);
    }

    if (response.reasoning_steps) {
      console.log('\n🧠 Reasoning Process:');
      response.reasoning_steps.forEach((step, i) => {
        console.log(`   ${i + 1}. ${step}`);
      });
    }
  } catch (error) {
    console.error(`❌ Error: ${error.message}`);
  }

  console.log();

  // Example 2: Domain specialist (Medical)
  console.log('='.repeat(60));
  console.log('Example 2: Medical Domain Specialist');
  console.log('='.repeat(60));

  const medicalQuestion = 'What are the symptoms of Type 2 diabetes?';
  console.log(`🏥 Medical Question: ${medicalQuestion}`);
  console.log();

  try {
    const response = await client.chat({
      message: medicalQuestion,
      namespace: 'medical'
    });
    console.log(`🤖 AetherMind (Medical): ${response.answer.substring(0, 300)}...`);
  } catch (error) {
    console.error(`❌ Error: ${error.message}`);
  }

  console.log();

  // Example 3: Memory search
  console.log('='.repeat(60));
  console.log('Example 3: Infinite Memory Search');
  console.log('='.repeat(60));

  const searchQuery = 'AGI capabilities';
  console.log(`🔍 Searching memory for: ${searchQuery}`);
  console.log();

  try {
    const memories = await client.searchMemory({
      query: searchQuery,
      topK: 3,
      includeEpisodic: true
    });

    if (memories && memories.length > 0) {
      console.log(`📚 Found ${memories.length} relevant memories:`);
      memories.forEach((memory, i) => {
        console.log(`\n   ${i + 1}. ${memory.text.substring(0, 200)}...`);
        console.log(`      Score: ${memory.score.toFixed(2)} | Time: ${memory.timestamp}`);
      });
    } else {
      console.log('   No memories found (try chatting first!)');
    }
  } catch (error) {
    console.error(`❌ Error: ${error.message}`);
  }

  console.log();

  // Example 4: Available namespaces
  console.log('='.repeat(60));
  console.log('Example 4: Available Knowledge Domains');
  console.log('='.repeat(60));

  try {
    const namespaces = await client.listNamespaces();
    console.log('🎯 Available domain specialists:');
    namespaces.forEach(ns => {
      console.log(`   - ${ns}`);
    });
  } catch (error) {
    console.log(`⚠️  Namespaces unavailable: ${error.message}`);
  }

  console.log();

  // Example 5: Usage statistics
  console.log('='.repeat(60));
  console.log('Example 5: Usage & Rate Limits');
  console.log('='.repeat(60));

  try {
    const usage = await client.getUsage();
    console.log('📊 API Usage Statistics:');
    console.log(`   Plan: ${usage.plan || 'N/A'}`);
    console.log(`   Requests Remaining: ${usage.requests_remaining || 'N/A'}`);
    console.log(`   Total Tokens Used: ${usage.total_tokens || 'N/A'}`);
    console.log(`   Reset At: ${usage.reset_at || 'N/A'}`);
  } catch (error) {
    console.log(`⚠️  Usage stats unavailable: ${error.message}`);
  }

  console.log();
  console.log('='.repeat(60));
  console.log('✨ Demo Complete!');
  console.log('='.repeat(60));
  console.log();
  console.log('📖 Learn more:');
  console.log('   - Documentation: https://aethermind.ai/documentation');
  console.log('   - GitHub: https://github.com/United-Visions/AetherAGI');
  console.log('   - Discord: https://discord.gg/aethermind');
  console.log();
  console.log('🚀 Ready to build with real AGI? Get your API key at:');
  console.log('   https://aethermind.ai');
  console.log();
}

// Run the demo
main().catch(error => {
  console.error('Fatal error:', error);
  process.exit(1);
});

File: sdk/examples/hello_aethermind.py
#!/usr/bin/env python3
"""
Hello AetherMind! 🚀
Simple example demonstrating AetherMind AGI integration
"""

import os
from aethermind import AetherMindClient

def main():
    print("=" * 60)
    print("🧠 Hello AetherMind - Python SDK Demo")
    print("=" * 60)
    print()
    
    # Get API key from environment or use demo
    api_key = os.getenv("AETHERMIND_API_KEY", "am_live_demo_key")
    
    if api_key == "am_live_demo_key":
        print("⚠️  Using demo API key. Set AETHERMIND_API_KEY environment variable")
        print("   to use your own key from https://aethermind.ai")
        print()
    
    # Initialize client
    print("📡 Connecting to AetherMind AGI...")
    try:
        client = AetherMindClient(api_key=api_key)
        print("✅ Connected!\n")
    except Exception as e:
        print(f"❌ Connection failed: {e}")
        return
    
    # Example 1: Basic chat
    print("=" * 60)
    print("Example 1: Basic Chat")
    print("=" * 60)
    
    question = "What is the difference between AI and AGI?"
    print(f"💬 Question: {question}")
    print()
    
    try:
        response = client.chat(question)
        print(f"🤖 AetherMind: {response['answer']}")
        
        if response.get('confidence'):
            print(f"📊 Confidence: {response['confidence']:.0%}")
        
        if response.get('reasoning_steps'):
            print(f"\n🧠 Reasoning Process:")
            for i, step in enumerate(response['reasoning_steps'], 1):
                print(f"   {i}. {step}")
    
    except Exception as e:
        print(f"❌ Error: {e}")
    
    print()
    
    # Example 2: Domain specialist (Legal)
    print("=" * 60)
    print("Example 2: Legal Domain Specialist")
    print("=" * 60)
    
    legal_question = "What are the key elements of a valid contract?"
    print(f"⚖️  Legal Question: {legal_question}")
    print()
    
    try:
        response = client.chat(legal_question, namespace="legal")
        print(f"🤖 AetherMind (Legal): {response['answer'][:300]}...")
    except Exception as e:
        print(f"❌ Error: {e}")
    
    print()
    
    # Example 3: Memory search
    print("=" * 60)
    print("Example 3: Infinite Memory Search")
    print("=" * 60)
    
    search_query = "AGI capabilities"
    print(f"🔍 Searching memory for: {search_query}")
    print()
    
    try:
        memories = client.search_memory(
            query=search_query,
            top_k=3,
            include_episodic=True
        )
        
        if memories:
            print(f"📚 Found {len(memories)} relevant memories:")
            for i, memory in enumerate(memories, 1):
                print(f"\n   {i}. {memory['text'][:200]}...")
                print(f"      Score: {memory['score']:.2f} | Time: {memory['timestamp']}")
        else:
            print("   No memories found (try chatting first!)")
    
    except Exception as e:
        print(f"❌ Error: {e}")
    
    print()
    
    # Example 4: Usage statistics
    print("=" * 60)
    print("Example 4: Usage & Rate Limits")
    print("=" * 60)
    
    try:
        usage = client.get_usage()
        print(f"📊 API Usage Statistics:")
        print(f"   Plan: {usage.get('plan', 'N/A')}")
        print(f"   Requests Remaining: {usage.get('requests_remaining', 'N/A')}")
        print(f"   Total Tokens Used: {usage.get('total_tokens', 'N/A')}")
        print(f"   Reset At: {usage.get('reset_at', 'N/A')}")
    except Exception as e:
        print(f"⚠️  Usage stats unavailable: {e}")
    
    print()
    print("=" * 60)
    print("✨ Demo Complete!")
    print("=" * 60)
    print()
    print("📖 Learn more:")
    print("   - Documentation: https://aethermind.ai/documentation")
    print("   - GitHub: https://github.com/United-Visions/AetherAGI")
    print("   - Discord: https://discord.gg/aethermind")
    print()
    print("🚀 Ready to build with real AGI? Get your API key at:")
    print("   https://aethermind.ai")
    print()

if __name__ == "__main__":
    main()
